src/components/molecules/LiveSchedule/LiveSchedule.tsx

```diff
@@ -0,0 +1,86 @@
+import { useSelector } from "hooks/useSelector";
+import { useUser } from "hooks/useUser";
+import React, { FC } from "react";
+import { VenueEvent } from "types/VenueEvent";
+import { isEventLive } from "utils/event";
+import { currentTimeInUnixEpoch, formatHourAndMinute } from "utils/time";
+import { isExternalUrl } from "utils/url";
+import { enterRoom } from "utils/useLocationUpdateEffect";
+import "./LiveSchedule.scss";
+
+const LiveSchedule: FC = () => {
+  const { user, profile } = useUser();
+  const { venueEvents, venue } = useSelector((state) => ({
+    venueEvents: state.firestore.ordered.venueEvents,
+    venue: state.firestore.ordered.currentVenue[0],
+  }));
+
+  const liveEvents =
+    venueEvents && venueEvents.length
+      ? venueEvents.filter((event) => isEventLive(event))
+      : [];
+
+  const onEventClick = (event: VenueEvent) => {
+    const room = venue?.rooms?.find((room) => room.title === event.room);
+
+    const isExternal = isExternalUrl(room.url);
+    if (isExternal) {
+      window.open(room.url, "_blank", "noopener,noreferrer");
+    } else {
+      window.location.href = room.url;
+    }
+    enterRoom(
+      user!,
+      { [`${venue.name}/${room.title}`]: currentTimeInUnixEpoch },
+      profile?.lastSeenIn
+    );
+  };
```

Comment:
As above RE: `useMemo`

Also, is there a better pattern than using `window` directly here? Eg. the hooks that come with React Router or similar?

- https://reactrouter.com/web/api/Hooks

Reply:
memoisation fixed in 77fc5d6d1829a42da2152d00871cde0b6f47269d

`window` usage is something we already do in many places, so can be cleaned up all at once in a future PR when we devise a better pattern.

Resolving.